blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -0,0 +1,655 @@
+   */
+  createNextBtn_() {
+    return this.createBtn_('blockly-next-btn', 'Find next');
+  }
+
+  /**
+   * Creates the button used to get the previous block in the list.
+   * @return {!HTMLButtonElement} The previous button.
+   * @protected
+   */
+  createPreviousBtn_() {
+    return this.createBtn_('blockly-previous-btn', 'Find previous');
+  }
+
+  /**
+   * Creates the button used for closing the search bar.
+   * @return {!HTMLElement} A button for closing the search bar.
+   * @protected
+   */
+  createCloseBtn_() {
+    return this.createBtn_('blockly-close-btn', 'Close search bar');
+  }
+
+  /**
+   * Creates a button for the workspace search bar.
+   * @param {string} className The class name for the button.
+   * @param {string} text The text to display to the screen reader.
+   * @return {!HTMLButtonElement} The created button.
+   * @private
+   */
+  createBtn_(className, text) {
+    // Create the button
+    const btn = document.createElement('button');
+    Blockly.utils.dom.addClass(btn, className);
+    btn.setAttribute('aria-label', text);
+    return btn;
+  }
+
+  /**
+   * Add event listener for clicking and keydown on the given button.
+   * @param {!HTMLButtonElement} btn The button to add the event listener to.
+   * @param {!Function} onClickFn The function to call when the user clicks on 
+   *     or hits enter on the button.
+   * @private
+   */
+  addBtnListener_(btn, onClickFn) {
+    btn.addEventListener('click', onClickFn);
+    // TODO: Review Blockly's key handling to see if there is a way to avoid
+    //  needing to call stopPropogation().
+    btn.addEventListener('keydown', e => {
+      if (e.key === "Enter") {
+        onClickFn(e);
+        e.preventDefault();  
+      } else if (e.key === "Escape") {
+        this.close();
+      }
+      e.stopPropagation();
+    });
+  }
+
+  /**
+   * Positions the search bar based on where the workspace's toolbox is.
+   * @protected
+   */
+  positionSearchBar_() {
+    // TODO: Handle positioning search bar when window is resized.
+    const metrics = this.workspace_.getMetrics();
+    if (this.workspace_.RTL) {
+      this.htmlDiv_.style.left = metrics.absoluteLeft + 'px';
+    } else {
+      if (metrics.toolboxPosition == Blockly.TOOLBOX_AT_RIGHT) {
+        this.htmlDiv_.style.right = metrics.toolboxWidth + 'px';
+      } else {
+        this.htmlDiv_.style.right = '0';
+      }
+    }
+    this.htmlDiv_.style.top = metrics.absoluteTop + 'px';
+  }
+
+  /**
+   * Handles input value change in search bar.
+   * @private
+   */
+  onInput_() {
+    if (this.searchOnInput) {
+      const inputValue = this.inputElement_.value.trim();
+      if (inputValue !== this.searchText_) {
+        this.searchAndHighlight(inputValue, this.preserveSelected);
+      }
+    }
+  }
+
+  /**
+   * Handles a key down for the search bar.
+   * @param {KeyboardEvent} e The key down event.
+   * @private
+   */
+  onKeyDown_(e) {
+    if (e.key === 'Escape') {
```

Comment:
Key is "Esc" instead of "Escape" on IE11. I assume you wanted to avoid keyCode because its deprecated. https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#Browser_compatibility

I think since Blockly uses keyCode, we could continue to use it here. Alternately, you could check for both e.key == 'Esc' and 'Escape', although I don't know what other non-standard key values IE uses, so it might be best to just go with keyCode to be on the safe side.

Reply:
updated to use keycode.